packages/daimo-api/src/contract/noteIndexer.ts

```diff
@@ -1,153 +1,136 @@
-            `bad NoteRedeemed, wrong amount: ${ephemeralOwner} ${logInfo()}`
-          );
-        }
-
-        // Mark as redeemed
-        note.status = redeemer === from ? "cancelled" : "claimed";
-        note.claimer = await this.nameReg.getEAccount(redeemer);
+    const ls = this.listeners;
+    console.log(`[NOTE] ${logs.length} logs for ${ls.length} listeners`);
+    ls.forEach((l) => l(logs));
+  }
 
-        opLogs.push({ type: "claim", noteStatus: { ...note } });
-      } else {
-        throw new Error(`unexpected event selector: ${selector}`);
+  private async loadCreated(
+    pg: Pool,
+    from: bigint,
+    to: bigint
+  ): Promise<DaimoNoteStatus[]> {
+    const result = await pg.query(
+      `
+        select
+        block_num,
+        encode(block_hash, 'hex') as block_hash,
+        tx_idx,
+        encode(tx_hash, 'hex') as tx_hash,
+        log_idx,
+        encode(log_addr, 'hex') as log_addr,
+
+        encode(f, 'hex') as "from",
+        encode(ephemeral_owner, 'hex') as "ephemeral_owner",
+        amount
+      from note_created
+      where block_num >= $1 and block_num <= $2
+    `,
+      [from, to]
+    );
+    const logs = result.rows.map(async (row) => {
+      console.log(`[NOTE] NoteCreated ${row.ephemeral_owner}`);
+      if (this.notes.get(row.ephemeral_owner) != null) {
+        throw new Error(
+          `dupe NoteCreated: ${row.ephemeral_owner} ${row.tx_hash} ${row.log_idx}`
+        );
       }
-    }
+      const sender = await this.ag.getEAccount(row.from);
+      const dollars = amountToDollars(BigInt(row.amount));
+      const newNote: DaimoNoteStatus = {
+        status: "confirmed",
+        dollars,
+        link: {
+          type: "note",
+          previewSender: getEAccountStr(sender),
+          previewDollars: dollars,
+          ephemeralOwner: row.ephemeral_owner,
+        },
+        sender,
+      };
+      this.notes.set(row.ephemeral_owner, newNote);
+      return newNote;
+    });
+    return await Promise.all(logs);
+  }
 
-    // Finally, invoke listeners to send notifications etc.
-    const ls = this.listeners;
-    console.log(`[NOTE] ${opLogs.length} logs for ${ls.length} listeners`);
-    ls.forEach((l) => l(opLogs));
-  };
+  private async loadRedeemed(
+    pg: Pool,
+    from: bigint,
+    to: bigint
+  ): Promise<DaimoNoteStatus[]> {
+    const result = await pg.query(
+      `
+        select
+        block_num,
+        encode(block_hash, 'hex') as block_hash,
+        tx_idx,
+        encode(tx_hash, 'hex') as tx_hash,
+        log_idx,
+        encode(log_addr, 'hex') as log_addr,
+
+        encode(f, 'hex') as "from",
+        encode(redeemer, 'hex') as "redeemer",
+        encode(ephemeral_owner, 'hex') as "ephemeral_owner",
+        amount,
+      from note_redeemed
+      where block_num >= $1 and block_num <= $2
+    `,
+      [from, to]
+    );
+    const logs = result.rows.map(async (row) => {
+      console.log(`[NOTE] NoteRedeemed ${row.ephemeral_owner}`);
```

Comment:
I think this is achieved in the calling function: https://github.com/daimo-eth/daimo/blob/048b95521644e43e3b8fbac89e5dd4b5bb9e5137/packages/daimo-api/src/contract/noteIndexer.ts#L29-L37

Reply:
I assume the dependency here is the coordination around `private notes: Map<Address, DaimoNoteStatus> = new Map();`. Since loadCreated is setting data on this map prior to loadRedeemed getting data I think we are good.